pkg/rpc/server/server.go

```diff
@@ -270,17 +270,39 @@ func (s *Server) handleHTTPRequest(w http.ResponseWriter, httpRequest *http.Requ
 		return
 	}
 
-	err := req.DecodeData(httpRequest.Body)
+	err := in.DecodeData(httpRequest.Body)
 	if err != nil {
-		s.writeHTTPErrorResponse(req, w, response.NewParseError("Problem parsing JSON-RPC request body", err))
+		s.writeHTTPErrorResponse(request.NewRequest(), w, response.NewParseError("Problem parsing JSON-RPC request body", err))
 		return
 	}
 
-	resp := s.handleRequest(req, nil)
-	s.writeHTTPServerResponse(req, w, resp)
+	resp := s.handleIn(in, nil)
+	s.writeHTTPServerResponse(in, w, resp)
 }
 
-func (s *Server) handleRequest(req *request.In, sub *subscriber) response.Abstract {
+func (s *Server) handleIn(in *request.In, sub *subscriber) interface{} {
+	if in.Request != nil {
+		return s.handleRequest(in.Request, sub)
+	}
+	resp := make([]response.Abstract, len(in.Batch))
+	ch := make(chan struct{})
+	for i, req := range in.Batch {
+		go func(i int, req *request.Request) {
```

Comment:
No goroutines for this, please. We already have one per-client and I think it's more than enough, otherwise batching clients would get disproportionate amount of CPU for their requests.